### PR TITLE
Installation of local flatpak files

### PR DESCRIPTION
**Pull request by isantop**
Adds a Flatpak Installer applet that allows installation of locally downloaded flatpakref files (such as those downloaded from Flathub or similar). This can be accessed through a button in the Flatpak tab, via cli with the `flatpak-installer` command, and via file manager by double-clicking on a FlatpakRef file. 

If the installation is successful, a notification letting the user know the process is complete is displayed. If it is unsuccessful, then an error dialog is displayed explaining the problem and no notification is displayed. 

The flatpak installer will add a remote for the flatpakref file to install it. This is required because the flatpak ref file doesn't contain the actual application, which needs to be downloaded from the remote. It is because of this that the installer will require an internet connection to install a flatpakref. This behavior matches that of the `flatpak` cli. 

The "Flatpak Installer" Open With item only appears as an option for flatpakref files. If the `flatpak-installer` command is used with a non-flatpakref file as an argument, it will present an error to the user explaining that the given file doesn't appear to be a valid file to install.

### CODE REVIEW

**Review by n3m0-22 — DISMISSED** [review]
This looks good.

Tested and working:
 - [x] Install flatpak using `.flatpakref` files (e.g., `org.ked.kdenlive.flatpakref`, `org.signal.Signal.flatpakref`, `se.sjoerd.DatMan.flatpakref`)
   - [x] Install `.flatpakref` using the new button in repoman 
   - [x] Install `.flatpakref` using flatpak-installer 
   - [x] Install `.flatpakref` using double click
 - [x] New Flatpak remote added
 - [x] Message showing the flatpak install was successfully for all the above
 - [x] Change `.flatpakref` to `.flatpak` or remove `.flatpakref` completely
   - [x] Creates an error trying to install from repoman, cli, and double clicking
 - [x] Alter `.flatpakref` file to break it and then try installing from repoman, cli, and double clicking
   - [x] All created an error 
   - [x] Different and usable errors depending on how the file was changed  
 - [x] Tested normal flatpak installs and searches with no regressions (with and without other sources added e.g., KDE Nightly) 
 - [x] No regressions in standard repoman testing
 
 This is not breaking and it does create an error when trying to install, but you can create a blank file `test.flatpakref` or even just `test` with no extension and if the file contains nothing but the text `[Flatpak Ref]` it will still show `Open With Flatpak Installer`.

Assuming that this is okay, this is ready. I'm approving now on the condition that this is acceptable behavior.

**Review by mmstick — DISMISSED** [review]

**Comment by isantop** [comment]
@n3m0-22 

> This is not breaking and it does create an error when trying to install, but you can create a blank file test.flatpakref or even just test with no extension and if the file contains nothing but the text [Flatpak Ref] it will still show Open With Flatpak Installer.

I believe that's due to Nautilus's MIME type matching, where theoretically that is a valid flatpakref file even if it doesn't contain any usable data. As long as it shows an error message, I think that would be acceptable behavior.

**Comment by n3m0-22** [comment]
@isantop I noticed one other thing. If you install the `flatpackref` using the cli the window that pops up does not grab the keyboard for the `Install` button. You need to hit `tab` a couple times to use `enter` to click `Install`. Using the mouse is fine.

**Comment by isantop** [comment]
> @isantop I noticed one other thing. If you install the `flatpackref` using the cli the window that pops up does not grab the keyboard for the Install button. You need to hit `tab` a couple times to use `enter` to click `Install`. Using the mouse is fine.

I'll set up the install button to be the default action

**Comment by isantop** [comment]
@n3m0-22 Just a quick note: 2d253c4 has the install button grab default, but it will only work in cases where the user opens the installer from the CLI or by double-clicking on a flatpakref file. If the user opens the installer via repoman, then they need to click the file chooser button to select the file to install, which sets focus on the file chooser button. GTK always sends the enter-key activate signal to the focused widget, and only sends it to the default widget if there is no widget with focus.

It is testing out well for CLI and open though.

**Review by n3m0-22 — DISMISSED** [review]
That's fixed it. I think it's fine using the mouse when installing from repoman it's having to go to the mouse when you are starting from the terminal that seemed off. This looks good to me.

**Comment by isantop** [comment]
I definitely agree, just wanted to make you aware of the caveat

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,10 @@
+repoman (2.4.0) UNRELEASED; urgency=medium
```
Comment (13r0ck) [inline debian/changelog]:
```suggestion
repoman (2.4.0) jammy; urgency=medium
```

**Review by 13r0ck — CHANGES_REQUESTED** [review]

**Comment by isantop** [comment]
@n3m0-22 The most recent commit shouldn't have changed anything with the code; it just updates the changelog to fix the issue mentioned above. I don't expect any breakage with testing

**Review by 13r0ck — APPROVED** [review]

**Review by n3m0-22 — APPROVED** [review]
Looks good.

**Pull request merged**